cl/recap/mergers.py

```diff
@@ -80,20 +80,27 @@ def find_docket_object(
     # pacer_case_id is required for Docket and Docket History uploads.
     d = None
     docket_number_core = make_docket_number_core(docket_number)
-    lookups = [
-        {
-            "pacer_case_id": pacer_case_id,
-            "docket_number_core": docket_number_core,
-        },
-        {"pacer_case_id": pacer_case_id},
-    ]
+    lookups = []
+    if pacer_case_id:
+        # Appellate RSS feeds don't contain a pacer_case_id, avoid lookups by
+        # blank pacer_case_id values.
+        lookups = [
+            {
+                "pacer_case_id": pacer_case_id,
+                "docket_number_core": docket_number_core,
+            },
+            {"pacer_case_id": pacer_case_id},
+        ]
     if docket_number_core:
         # Sometimes we don't know how to make core docket numbers. If that's
         # the case, we will have a blank value for the field. We must not do
         # lookups by blank values. See: freelawproject/courtlistener#1531
         lookups.append(
             {"pacer_case_id": None, "docket_number_core": docket_number_core},
         )
+        lookups.append(
+            {"docket_number_core": docket_number_core},
+        )
```

Comment:
Yeah about this questions:

> Why do we need this lookup? Earlier we match on pacer_case_id + docket_number_core, then we try on no pacer_case_id + docket_number_core. I guess this would match on docket_number_core with a mismatched pacer_case_id, which seems bad, doesn't it?

Yeah, I added this new lookup because I realized a case where for instance, the appellate docket is added first by the extension and they do have a `pacer_case_id` (like the new appellate dockets), then when the scraper starts scraping appellate courts and we get data for one of those dockets that have a `pacer_case_id`, since the RSS data don't contains a `pacer_case_id` the first three lookups wont' work:

```
{"pacer_case_id": pacer_case_id, "docket_number_core": docket_number_core,}
{"pacer_case_id": pacer_case_id}
{"pacer_case_id": None, "docket_number_core": docket_number_core,}
```

Since we don't have the `pacer_case_id` in the RSS data and also the `pacer_case_id` is not `None` in the docket.
So in those cases a new docket will be created instead of updating the existing one.

That's why I added a lookup only using the `docket_number_core` so that the docket can be found.
This example is tested in: `test_appellate_merge_rss_feed_with_case_id`

But yes, as you mention this might be risky so a docket with the wrong `pacer_case_id` can be matched since we don't have the `pacer_case_id` to compare it. 
Would it help if we just apply this lookup for appellate dockets? 
In #2511 we do an additional check confirming the `docket_number` but not sure if that would be enough.

> Assuming there's a logical answer for the first question, why do these as two different appends rather than just doing one?

Changed, now using extend to add the lookups.

Reply:
Got it. Hopefully this should be OK. We'll see, I guess.